docs/how_to_dev_frontend_en.md

```diff
@@ -0,0 +1,105 @@
+# VisualDL Web App Develop
+
+VisualDL has two major components. One is the Python/C++ SDK that logs the matrix during training,
+the other is the web-app that display visualized training data.
+
+The web-app reads the extract data and draw scalars graph or histograms graphs.
+The web-app is where VisualDL transforms the code data into graphs so users can visualize the training progress and make a better network design evaluation.
+
+This article will go over the basic web-app architecture and development steps.
+
+## Web App Overview
+The VisualDL web-app home page is constructed by three major Javascript files
+1. manifest.js: The manifest configures basic settings  
+1. vendor.js: The vendor.js stores all the 3rd party libraries.
+1. index.js: The index.js contains all the graph algorithms. All navigations, setting panels are implemented here.
+It finds the 'root' DOM in the index.html and attaches the web-app to it.
+
+The graph algorithms are defined in the `.san` files. san is the ECharts' graph framework. It can pilot charts and graphs with just a few lines of code.
+To learn more about ECharts's framework, please visit [ECharts](https://ecomfe.github.io/echarts-doc/public/en/index.html).
+
```

Comment:
There is a lot of other complexity that can be introduced here, such as how these components come together.